File: src/Pubsub.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import Float64
from std_msgs.msg import String

global resultado
resultado = 0

def funcion(msg): # Tipo callback
  global resultado
  resultado = msg.data  

if __name__ == "__main__":

  rospy.init_node('publishersubscriber')
  rospy.Subscriber('counter', Float64, funcion)
  pub = rospy.Publisher('counter_pair', String, queue_size=1)
  rate = rospy.Rate(2) # 2 Hz
  
  while not rospy.is_shutdown():
    
    message = String()
    
    if (resultado % 2 == 0):
      message.data = "Valor par"
    else:
      message.data = "Valor impar"
    
    pub.publish(message)
    rate.sleep()

File: src/Pubsub.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include "std_msgs/Float64.h"
#include <sstream>

int result = 0;

void function_cb(const std_msgs::Float64::ConstPtr& msg)
{
  result = msg->data;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "publishersubscriber");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("counter", 1, function_cb);
  ros::Publisher pub = n.advertise<std_msgs::String>("counter_pair", 1);
  ros::Rate loop_rate(2);

  int count = 0;
  while (ros::ok())
  {
    std_msgs::String message;
    
    if(result % 2 == 0){
      message.data = "Pair value";
    }
    else{
      message.data = "Odd value";
    }
    
    pub.publish(message);
    ros::spinOnce();
    loop_rate.sleep();
    ++count;
  }
  return 0;
}
